src/Revit/RevitHyparTools/WallByProfile.cs

```diff
@@ -1,40 +1,34 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
-using Elements;
 using Elements.Geometry;
 using Elements.Geometry.Solids;
 
-namespace Hypar.Revit {
-    [UserElement]
-    public class WallByProfile : Wall
+namespace Elements {
+    public partial class WallByProfile 
     {
-        public Line CenterLine {get;}
-        public double Thickness {get;}
-        public WallByProfile(Line centerLine,
-                             Profile profile,
-                             double thickness,
-                             Material material = null,
-                             Transform transform = null,
-                             Representation representation = null,
-                             bool isElementDefinition = false,
-                             Guid id = default(Guid),
-                             string name = null) : base(transform != null ? transform : new Transform(),
-                                                 material != null ? material : BuiltInMaterials.Concrete,
-                                                 representation != null ? representation : new Representation(new List<SolidOperation>()),
-                                                 isElementDefinition,
-                                                 id != default(Guid) ? id : Guid.NewGuid(),
-                                                 name)
-        {
-            Profile = profile;
-            CenterLine = centerLine;
-            Thickness = thickness;
+        public WallByProfile(Profile @profile, double @thickness, Line @centerline, Transform @transform=null, Material @material=null, Representation @representation=null, bool @isElementDefinition=false)
+            : base(transform != null ? transform : new Transform(),
+                   material != null ? material : BuiltInMaterials.Default,
+                   representation != null ? representation: new Representation(new List<SolidOperation>()),
+                   isElementDefinition,
+                   Guid.NewGuid() ,
+                   "Wall by Profile") {
+
+            this.Profile = @profile;
+            this.Thickness = @thickness;
+            this.Centerline = @centerline;
         }
 
         public override void UpdateRepresentations() {
             this.Representation.SolidOperations.Clear();
-            // The wall is a simple extrusion of the profile
-            this.Representation.SolidOperations.Add(new Extrude(this.Profile, this.Thickness, Profile.Perimeter.Normal(), false));
+
+            // to ensure the correct direction, we find the direction form a point on the polygon to the vertical plane of the centerline
+            var point = Profile.Perimeter.Vertices.First();
```

Comment:
I will require good excuses to introduce Linq stuff like `First`, if only because Linq tends to be abused. Is there a reason we can't do `Profile.Perimeter.Vertices[0]`?

Reply:
what do you mean Linq tends to be abused?  I think it's concise, clear and a pleasure to use.  what are you looking out for?
Of course changing to getting by index will work just as well.

Reply:
Let's leave this usage for now. A conversation lazy evaluation is outside the scope of this pr.